cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/RawComparatorWrapper.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.internal.app.runtime.batch;
+
+import org.apache.hadoop.conf.Configurable;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.io.RawComparator;
+import org.apache.hadoop.mapreduce.Job;
+import org.apache.hadoop.mapreduce.MRJobConfig;
+
+/**
+ * Wraps user-defined implementation of {@link RawComparator} class which allows perform extra configuration,
+ * such as initialization with MapReduceTaskContext, if the delegate class implements ProgramLifeCycle.
+ */
+abstract class RawComparatorWrapper implements RawComparator, Configurable {
+
+  private RawComparator delegate;
+  private Configuration conf;
+
+  abstract String getDelegateClassAttr();
+
+  @Override
+  public void setConf(Configuration conf) {
+    this.conf = conf;
+    this.delegate = WrapperUtil.createDelegate(conf, getDelegateClassAttr());
+  }
+
+  @Override
+  public Configuration getConf() {
+    return conf;
```

Comment:
Same as for the partitioner wrapper. If the delegate is Configurable(), then we should delegate this to it.